docs/keepalived.md

```diff
@@ -0,0 +1,96 @@
+# Purpose of keepalived implementation
+
+Adding support for [keepalive](http://www.keepalived.org/documentation.html) to enable high availability in an haproxy deployment, leveraging the https://en.wikipedia.org/wiki/Virtual_Router_Redundancy_Protocol more formally [RFC 5798](https://tools.ietf.org/html/rfc5798). See [keep-alived man page](https://linux.die.net/man/5/keepalived.conf) for more precision over capabilities of keep-alived as well as the [keep-alived user manual](http://www.keepalived.org/pdf/UserGuide.pdf)
+
+This enables declaring an virtual IP (``keepalived.vip``) that will automatically fail over between the multiple haproxy VMs: the master will initially be the bosh vm for haproxy job instance 0. The default IP addresses assigned by bosh to vms on eth0 are used within the VRRP protocol.
+
+Prereqs:
+ * The haproxy VMs must be within the same broadcast domain, i.e. receive multicast traffic sent to the 224.0.0.18 broadcast and IP protocol number 112.
+* The clients using this VIP must be within the [same broadcast domain](https://en.wikipedia.org/wiki/Broadcast_domain) as the haproxy vms and accepting ARP gratuitious
+
```

Comment:
yeah this won't work in aws, gcp, or azure; only vsphere most likely, as it requires broadcast.  there is a way to make it work with those clouds but you'd need to have forked scripts for each that use their CLI to rebind the public/elastic IP on failover, e.g. 

https://blog.logentries.com/2014/12/keepalived-and-haproxy-in-aws-an-exploratory-guide/

of course rebinds can take a couple of minutes so not quite "HA" but better than nothing ;)